main.tf

```diff
@@ -298,6 +301,26 @@ resource "google_alloydb_instance" "read_pool" {
     }
   }
 
+  depends_on = [google_alloydb_instance.primary]
+}
+
+resource "random_password" "passwords" {
```

Comment:
@viliampucik this will cause problem during failover as same module is use to create cross regional replica instance. follow this [doc](https://github.com/GoogleCloudPlatform/terraform-google-alloy-db/blob/main/examples/simple_example/README.md) for testing failover/switchover

Reply:
@imrannayer I see your point. Would you prefer to keep user management decoupled (outside the module), even for clusters without any cross regional replication?

Reply:
@viliampucik yes. There is always a chance someone will create a standalone cluster and create cross region cluster later.

Reply:
Understood. I will remove the user management part from the PR.